src/Deriving/DepTyCheck/Gen/ForOneTypeCon/Impl.idr

```diff
@@ -121,23 +120,23 @@ canonicConsBody sig name con = do
             ]
 
         -- Order pairs by the first element like they are present in the constructor's signature
-        orderLikeInCon : Foldable f => f (String, String) -> List (String, String)
+        orderLikeInCon : Foldable f => f (Name, Name) -> List (Name, Name)
         orderLikeInCon = do
           let conArgStrNames = mapMaybe argStrName con.args
```

Comment:
Then `mapMaybe name con.args` and just remove `argStrName` (its name now even became wrong)